### PR TITLE
Add Ruff linting to Runner

### PR DESCRIPTION
**Pull request by petechd**
### What is the context of this PR?
Replaces Flake8 linter with Ruff and its multiple plugins to improve the performance of linting and code quality. It is part of a Spike to investigate usage of Ruff in eq-questionnaire-runner.

### How to review
The base branch for this change is already on Python 3.12 so you will need to make sure 3.12 is used in your IDE as well.

### Checklist
* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: scripts/run_lint_python.sh [hunk 1]
```diff
@@ -28,9 +31,6 @@ display_result $? 1 "Flake 8 code style check"
 find . -type f -name "*.py" | xargs pylint --reports=n --output-format=colorized --rcfile=.pylintrc -j 0
 display_result $? 1 "Pylint linting check"
 
-isort --check .
```
Comment (liamtoozer) [inline scripts/run_lint_python.sh:31]: Is it worth removing this from the packages, as well as the others that ruff replaces?

File: scripts/run_lint_python.sh [hunk 2]
```diff
@@ -18,8 +18,11 @@ function display_result {
   fi
 }
 
-flake8 --max-complexity 10 --count
-display_result $? 1 "Flake 8 code style check"
+#flake8 --max-complexity 10 --count
+pwd
+ruff --version
```
[the remark this answers is not in the document]
Comment (petechd) [reply scripts/run_lint_python.sh]: No, left some of it for testing and experiments, will remove it now. Even the `--config` is not needed as it defaults to pyproject.toml anyway.

File: scripts/run_lint_python.sh (hunk 1 above)
Comment (petechd) [reply scripts/run_lint_python.sh:31]: I think I removed everything now, `pep8` is used by `pylint-mccabe` and pylint stays for now.

**Review by rmccar — APPROVED** [review]

**Review by liamtoozer — APPROVED** [review]
Great work!

**Pull request merged**